cdap-app-fabric/src/main/java/co/cask/cdap/operations/OperationalStatsReader.java

```diff
@@ -0,0 +1,109 @@
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.Map;
+import javax.management.AttributeNotFoundException;
+import javax.management.InstanceNotFoundException;
+import javax.management.IntrospectionException;
+import javax.management.MBeanAttributeInfo;
+import javax.management.MBeanException;
+import javax.management.MBeanInfo;
+import javax.management.MBeanServer;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+import javax.management.ReflectionException;
+
+/**
+ * Reads operational stats collected using {@link OperationalStatsReader}.
+ */
+public class OperationalStatsReader {
+
+  private static final Logger LOG = LoggerFactory.getLogger(OperationalStatsReader.class);
+
+  /**
+   * Gets stats for all services, of the specified statType.
+   *
+   * @param statType the type of stats to return
+   * @return a map from service name to a map containing the stats of the specified type for that service
+   */
+  public Map<String, Map<String, String>> getStatsOfType(String statType)
+    throws MalformedObjectNameException, IntrospectionException, InstanceNotFoundException, ReflectionException,
+    AttributeNotFoundException, MBeanException {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(statType), "Stat type should not be null or empty");
+    Hashtable<String, String> properties = new Hashtable<>();
+    // we want to fetch basic details for all services, so use * as the service name
+    properties.put(OperationalStatsConstants.SERVICE_NAME_KEY, "*");
+    // we only want to fetch basic stats, so only fetch stats of the type info
+    properties.put(OperationalStatsConstants.STAT_TYPE_KEY, statType);
+    ObjectName objectName = new ObjectName(OperationalStatsConstants.JMX_DOMAIN, properties);
+    Map<String, Map<String, String>> result = new HashMap<>();
+    MBeanServer mbs = ManagementFactory.getPlatformMBeanServer();
+    for (ObjectName name : mbs.queryNames(objectName, null)) {
+      String serviceName = name.getKeyProperty(OperationalStatsConstants.SERVICE_NAME_KEY);
+      MBeanInfo mBeanInfo = mbs.getMBeanInfo(name);
+      Map<String, String> stats = new HashMap<>();
+      for (MBeanAttributeInfo attributeInfo : mBeanInfo.getAttributes()) {
+        // casting to String is ok, because we know that all info stats are supposed to be strings.
+        stats.put(attributeInfo.getName().toLowerCase(), (String) mbs.getAttribute(name, attributeInfo.getName()));
+      }
+      result.put(serviceName, stats);
+      LOG.trace("Found object {} with attributes {}", name, stats);
+    }
+    return result;
+
+  }
+
+  /**
+   * Get the operational stats for a given service.
+   *
+   * @param serviceName the service for which operational stats are required
+   * @return a map of stat type (e.g. storage, compute, etc) to a map of stats of the specific type
+   */
+  public Map<String, Map<String, Object>> getOperationalStats(String serviceName)
```

Comment:
Seems like this and previous are quite the same, the only different is the call to `getKeyProperty`?